app/src/main/java/comcmput301f16t01/github/carrier/SetLocationsActivity.java

```diff
@@ -0,0 +1,332 @@
+            }
+        } else if(point.equals("start")) {
+            if(intent.hasExtra("startLocation")) {
+                locationPoint = new Gson().fromJson(bundle.getString("startLocation"), Location.class);
+            }
+            if(intent.hasExtra("endLocation")) {
+                lastBundle.putString("endLocation",bundle.getString("endLocation"));
+            }
+        }
+
+        checkPermissions();
+
+        // Create an instance of GoogleAPIClient.
+        if (googleApiClient == null) {
+            googleApiClient = new GoogleApiClient.Builder(activity)
+                    .addConnectionCallbacks((GoogleApiClient.ConnectionCallbacks) activity)
+                    .addOnConnectionFailedListener((GoogleApiClient.OnConnectionFailedListener) activity)
+                    .addApi(LocationServices.API)
+                    .build();
+        }
+
+        OpenStreetMapTileProviderConstants.setCachePath(new File(Environment.getExternalStorageDirectory().getPath() + "osmdroid2").getAbsolutePath());
+
+        MapView map = (MapView) findViewById(R.id.map);
+        map.setTileSource(TileSourceFactory.MAPNIK);
+        map.setBuiltInZoomControls(true);
+        map.setMultiTouchControls(true);
+
+        if(locationPoint != null) {
+            GeoPoint geoPoint = new GeoPoint(locationPoint.getLatitude(), locationPoint.getLongitude());
+            map = (MapView) findViewById(R.id.map);
+            setLocationMarker(map, geoPoint);
+            IMapController mapController = map.getController();
+            mapController.setZoom(18);
+            mapController.setCenter(geoPoint);
+        }
+    }
+
+    // the following two functions come from: https://developer.android.com/training/location/retrieve-current.html
+    protected void onStart() {
+        googleApiClient.connect();
+        super.onStart();
+    }
+
+    protected void onStop() {
+        googleApiClient.disconnect();
+        super.onStop();
+    }
+
+    /**
+     * Set the location marker on the map for the user-selected location
+     *
+     * @param map
+     * @param geoPoint
+     */
+    private void setLocationMarker(MapView map, GeoPoint geoPoint) {
+        final Marker startMarker = new Marker(map);
+        startMarker.setPosition(geoPoint);
+        startMarker.setAnchor(Marker.ANCHOR_CENTER, Marker.ANCHOR_BOTTOM);
+
+        startMarker.setDraggable(true);
+        startMarker.setOnMarkerDragListener(new Marker.OnMarkerDragListener() {
+            @Override
+            public void onMarkerDrag(Marker marker) {
+
+            }
+
+            @Override
+            public void onMarkerDragEnd(Marker marker) {
+                locationPoint.setLatitude(marker.getPosition().getLatitude());
+                locationPoint.setLongitude(marker.getPosition().getLongitude());
+            }
+
+            @Override
+            public void onMarkerDragStart(Marker marker) {
```

Comment:
OnMarkerDragListener is an abstract class. This method must exist but we don't need it to do anything.